### ISSUE TITLE
Bash too late in big git repositories

### ISSUE THREAD

**Issue by JulioSerna** [Vauxoo/docker-odoo-image#93]
In some repositories like this,
![bash odoo-8 0](https://cloud.githubusercontent.com/assets/7595505/14004101/5e77ecf6-f11d-11e5-9390-4bc005dfea51.png)

The bash color is causing  delay, and in that case the `odoo-8.0` always is red

**Comment by JulioSerna** [comment]
I propose only keep the bash color for the repositories build like this,
![bash-color-build](https://cloud.githubusercontent.com/assets/7595505/14004224/0c5222f6-f11e-11e5-8d53-d794cfcfef46.png)

**Comment by gomezvillegasdaniel** [comment]
+1

**Comment by gomezvillegasdaniel** [comment]
:+1:

**Comment by KarenKawaii** [comment]
:+1:

**Comment by zaoral** [comment]
I really like and use the bash color branch status. But if it is a performance issue I agree to disable it.

I vote for disable this as a default feature but I will like to have an extra command or an easy way to activate the feature in my containers.

**Comment by zaoral** [comment]
Why I support the color bash status for directory with branch? Because generally when we enter inside a directory that have a branch we intermediately do a `git status` to know what is the current status of the branch.

The colorized bash do exactly the same step: Check if the current directory is a branch, and if True make a `git status`: if there are changes turns the bash to red, If there is not change then turns the bash to green.

So the colorized bash is saving this first `git status` we do almost all the time.

I think the problem with the odoo-8.0 branch is that is too big to check the status. We need a way to optimize the branch. IMHO a way to do that is to compress the file revisions.  This can be accomplish with this command

```bash
git gc --aggressive
```

For more information about how this command works only need to make a `git gc --help` but here is a summary of the information we need.

```
    git-gc - Cleanup unnecessary files and optimize the local repository

   Runs a number of housekeeping tasks within the current repository, such as compressing file revisions (to reduce disk space and increase
      performance) and removing unreachable objects which may have been created from prior invocations of git add.

OPTIONS
       --aggressive
           Usually git gc runs very quickly while providing good disk space utilization and performance. This option will cause git gc to more
           aggressively optimize the repository at the expense of taking much more time. The effects of this optimization are persistent, so this option
           only needs to be used occasionally; every few hundred changesets or so.
```

I hope this information and this point of view help to make the decision.
How ever, if we still need to disable this functionality please make a way to manually re enable it for those who use it.

Best Regards

**Comment by osvalr** [comment]
as an alternative for this is using `zsh`, it is already installed

**Comment by moylop260** [comment]
@osvalr 

> as an alternative for this is using `zsh`, it is already installed

The main issue isn't the `/bin/bash` or `zsh`

We have the follow script to detect a change in the git repository and change the color of branch bash:
[git-branch-color-prompt](https://github.com/Vauxoo/docker-odoo-image/blob/b146ad2f8f27829ca2cba46b97858063eefd934d/odoo-shippable/files/bash_colors#L85-L99)

If we add the same script in `zsh` we will have the same issue.

Do you understand me?

**Comment by moylop260** [comment]
All
I added `git gc --aggressive` script to [Dockerfile](https://github.com/Vauxoo/docker-odoo-image/blob/b146ad2f8f27829ca2cba46b97858063eefd934d/odoo-shippable/Dockerfile#L128) and I executed `git pull vauxoo/odoo-80-image-shippable-auto` in developers servers (raditz).

I hope your feedback of performance to kwon the future of this feature.

Thanks!

**Comment by osvalr** [comment]
@moylop260 I see. I'll be watching how long it takes with that change you made.

**Comment by moylop260** [comment]
FYI `zsh` use follow [script](https://github.com/robbyrussell/oh-my-zsh/blob/c3b3de5f29b72b45393b62d794450ad3a72703a8/lib/git.zsh#L11-L30) and have the same issue with big git folders.

**Comment by LuisAlejandro** [comment]
@moylop260 please close this issue, as it was resolved by #93
